ninja-metrics-graphite/src/test/java/ninja/metrics/graphite/NinjaGraphiteTest.java

```diff
@@ -0,0 +1,93 @@
+import com.codahale.metrics.graphite.GraphiteReporter.Builder;
+import com.codahale.metrics.graphite.GraphiteSender;
+
+import ninja.metrics.MetricsService;
+import ninja.utils.NinjaPropertiesImpl;
+
+@PrepareForTest(GraphiteReporter.class)
+@RunWith(PowerMockRunner.class)
+public class NinjaGraphiteTest {
+
+    private static final String TEST_HOSTNAME = "test.hostname";
+    private static final String TEST_PREFIX = "my.prefix";
+
+    @Mock
+    private NinjaPropertiesImpl ninjaProperties;
+
+    @Mock
+    private MetricsService metricsService;
+
+    @Mock
+    private Builder builder;
+
+    private NinjaGraphite ninjaGraphite;
+
+    @Before
+    public void setup() {
+
+        // configure mocks
+        MockitoAnnotations.initMocks(this);
+        Mockito.when(metricsService.getHostname()).thenReturn(TEST_HOSTNAME);
+        Mockito.when(ninjaProperties.getBooleanWithDefault(Mockito.eq("metrics.graphite.enabled"), Mockito.anyBoolean())).thenReturn(Boolean.TRUE);
+        Mockito.when(ninjaProperties.getOrDie(Mockito.eq("metrics.graphite.address"))).thenReturn("localhost");
+        Mockito.when(ninjaProperties.getIntegerWithDefault(Mockito.eq("metrics.graphite.port"), Mockito.anyInt())).thenReturn(2003);
+        Mockito.when(ninjaProperties.getBooleanWithDefault(Mockito.eq("metrics.graphite.pickled"), Mockito.anyBoolean())).thenReturn(Boolean.FALSE);
+        Mockito.when(ninjaProperties.getWithDefault(Mockito.eq("metrics.graphite.period"), Mockito.anyString())).thenReturn("60s");
+
+        PowerMockito.mockStatic(GraphiteReporter.class);
+        PowerMockito.when(GraphiteReporter.forRegistry(Mockito.any())).thenReturn(builder);
+        Mockito.when(builder.prefixedWith(Mockito.anyString())).thenCallRealMethod();
+        Mockito.when(builder.convertRatesTo(Mockito.any())).thenCallRealMethod();
+        Mockito.when(builder.convertDurationsTo(Mockito.any())).thenCallRealMethod();
+        Mockito.when(builder.filter(Mockito.any())).thenCallRealMethod();
+        Mockito.when(builder.build(Mockito.any(GraphiteSender.class))).thenCallRealMethod();
+
+        // create instance under test
+        ninjaGraphite = new NinjaGraphite(ninjaProperties, metricsService);
+    }
+
+    @Test
+    public void testWithoutCustomPrefix() {
+        mockPrefix(null);
+        ninjaGraphite.start();
+        Mockito.verify(builder, Mockito.times(1)).prefixedWith(TEST_HOSTNAME);
+    }
+
+    @Test
+    public void testWithEmptyCustomPrefix() {
+        mockPrefix("");
+        ninjaGraphite.start();
+        Mockito.verify(builder, Mockito.times(1)).prefixedWith(TEST_HOSTNAME);
+    }
+
+    @Test
+
+    public void testWithCustomPrefix() {
+        mockPrefix(TEST_PREFIX);
+        ninjaGraphite.start();
+        Mockito.verify(builder, Mockito.times(1)).prefixedWith(TEST_PREFIX);
+    }
+
+    private void mockPrefix(String prefix) {
+        Mockito.when(ninjaProperties.get(Mockito.eq("metrics.graphite.prefix"))).thenReturn(prefix);
```

Comment:
Please add this inside the test methods and don't use a separate method. Usually I'd agree that a well named method makes a lot of sense - but because it is just one line it is better imho to move it directly into the test methods.